packages/dev/core/src/Collisions/pickingInfo.ts

```diff
@@ -105,15 +105,10 @@ export class PickingInfo {
             result = Vector3.Cross(p1p2, p3p2);
         }
 
-        // Flip the normal if the picking ray is in the same direction.
-        if (this.ray && Vector3.Dot(result, this.ray.direction) > 0) {
-            result.negateInPlace();
-        }
-
-        if (useWorldCoordinates) {
-            let wm = this.pickedMesh.getWorldMatrix();
+        const transformNormalToWorld = (n: Vector3) => {

```

Comment:
nit: if you pass in the pickedMesh as an arg, then you won't need the `!`
```suggestion
        const transformNormalToWorld = (pickedMesh: AbstractMesh, n: Vector3) => {
```